packages/client/hooks/useAnimatedMeetingCard.ts

```diff
@@ -0,0 +1,46 @@
+import {useEffect, useRef} from 'react'
+import requestDoubleAnimationFrame from '../components/RetroReflectPhase/requestDoubleAnimationFrame'
+import {BezierCurve} from '../types/constEnums'
+import {TransitionStatus} from './useTransition'
+
+const useAnimatedMeetingCard = (displayIdx: number, status: TransitionStatus) => {
+  const ref = useRef<HTMLDivElement>(null)
+  const lastIdxRef = useRef(displayIdx)
+  const lastRectRef = useRef({left: 0, top: 0})
+  // standard FLIP animation
+  useEffect(() => {
+    const idx = lastIdxRef.current
+    const rect = lastRectRef.current
+    const el = ref.current
+    if (!el) return
+    // offset measurements are taken after a DOM paint
+    const {offsetTop, offsetLeft, style} = el
+    if (idx !== displayIdx) {
+      // interpolate last - first in FLIP
+      const diffLeft = rect.left - offsetLeft
+      const diffTop = rect.top - offsetTop
+      const duration = 350
+      // move it back where it was
+      style.transform = `translate(${diffLeft}px,${diffTop}px)`
+      requestDoubleAnimationFrame(() => {
```

Comment:
`requestAnimationFrame` tells the browser that we want to perform an animation before the next repaint. `requestDoubleAnimationFrame` is required here because it calls any animations that might block the main thread which would lead to jankiness?

Reply:
it's _almost_ the same as a setTimeout of 0. when you assign a value to `style.transform` it queues up that change to be done on the next frame, but sometimes it can happen in the same frame. then to reset the value we want to wait another frame so it doesn't undo the change before it has started.